app/services/collators/gross_income_collator.rb

```diff
@@ -17,106 +18,93 @@ def initialize(assessment:, submission_date:, employments:, disposable_income_su
+               end
+
+      @disposable_income_summary.update!(employment_income_deductions: result.employment_income_deductions,
+                                         fixed_employment_allowance: result.fixed_employment_allowance,
+                                         tax: result.tax,
+                                         national_insurance: result.national_insurance)
+      add_remarks if @employments.count > 1
+
+      result
+    end
+
     def add_remarks
       my_remarks = @assessment.remarks
-      my_remarks.add(:employment, :multiple_employments, employment_client_ids)
+      my_remarks.add(:employment, :multiple_employments, @employments.map(&:client_id))
       @assessment.update!(remarks: my_remarks)
     end
 
-    def employment_client_ids
-      @employments.map(&:client_id)
-    end
-
     def income_categories
       CFEConstants::VALID_INCOME_CATEGORIES.map(&:to_sym)
     end
 
-    def populate_attrs
-      attrs = default_attrs
-
-      income_categories.each do |category|
-        populate_income_attrs(attrs, category)
+    def perform_collation(employment_income_subtotals)
+      regular_income_categories = income_categories.map do |category|
+        calculate_category_subtotals(category)
       end
 
-      if @employments.count < 2
-        attrs[:total_gross_income] += @gross_income_summary[:gross_employment_income]
-      end
-
-      attrs
-    end
+      total_gross_income = employment_income_subtotals.gross_employment_income +
+        regular_income_categories.sum(&:all_sources) +
+        monthly_student_loan +
+        monthly_unspecified_source
 
-    def populate_income_attrs(attrs, category)
-      attrs[:"#{category}_bank"] = category == :benefits ? monthly_state_benefits : categorised_income[category]
-      attrs[:"#{category}_cash"] = monthly_cash_transaction_amount_by(gross_income_summary: @gross_income_summary, operation: :credit, category:)
-      attrs[:"#{category}_all_sources"] = attrs[:"#{category}_bank"] + attrs[:"#{category}_cash"]
-      attrs[:total_gross_income] += attrs[:"#{category}_all_sources"]
-    end
-
-    def default_attrs
-      # setup initial values here, populate_income_attrs above adds the other income(s)
-      # to the default start values here
-      {
-        total_gross_income: monthly_student_loan + monthly_unspecified_source,
+      PersonGrossIncomeSubtotals.new(
+        total_gross_income:,
         monthly_student_loan:,
         monthly_unspecified_source:,
-        student_loan: categorised_income[:student_loan],
-        unspecified_source: categorised_income[:unspecified_source],
-        monthly_other_income: categorised_income[:total],
-        monthly_state_benefits:,
-      }
+        regular_income_categories:,
+        employment_income_subtotals:,
+      )
+    end
+
+    def calculate_category_subtotals(category)
+      bank = category == :benefits ? monthly_state_benefits : categorised_bank_transactions[category]
```

Comment:
can we have some bracket here:  bank = (category == :benefits) ? monthly_state_benefits : categorised_bank_transactions[category] for readability?

Reply:
Rubocop says no! `Omit parentheses for ternary conditions. (convention:Style/TernaryParentheses)`. But I take your point re readability so I'll turn it into a multi-line if statement instead.